pandas/core/internals/blocks.py

```diff
@@ -1341,6 +1341,25 @@ def shift(self, periods: int, axis: int = 0, fill_value=None):
 
         return [self.make_block(new_values)]
 
+    def _maybe_reshape_where_args(self, values, other, cond, axis):
+        transpose = self.ndim == 2
+
+        cond = _extract_bool_array(cond)
+
+        # If the default broadcasting would go in the wrong direction, then
+        # explicitly reshape other instead
+        if getattr(other, "ndim", 0) >= 1:
+            if values.ndim - 1 == other.ndim and axis == 1:
+                other = other.reshape(tuple(other.shape + (1,)))
+            elif transpose and values.ndim == self.ndim - 1:
+                # TODO(EA2D): not neceesssary with 2D EAs
+                cond = cond.T
+
+        if not hasattr(cond, "shape"):
+            raise ValueError("where must have a condition that is ndarray like")
```

Comment:
is this actually hit? shouldn't this be an assertion (i know this is friendler of course), but still.

Reply:
actually you explicity check this on L1347 no?

Reply:
good catch, will remove

Reply:
updated + green